yb-voyager/src/queryparser/traversal_plpgsql.go

```diff
@@ -0,0 +1,171 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package queryparser
+
+import (
+	"strings"
+)
+
+const (
+	PLPGSQL_EXPR = "PLpgSQL_expr"
+	QUERY        = "query"
+)
+
+/*
+Query example-
+
+	CREATE OR REPLACE FUNCTION func_example_advi(
+	   sender_id INT,
+	   receiver_id INT,
+	   transfer_amount NUMERIC
+
+)
+RETURNS VOID
+LANGUAGE plpgsql
+AS $$
+BEGIN
+
+	-- Acquire advisory locks to prevent concurrent updates on the same accounts
+	PERFORM pg_advisory_lock(sender_id);
+	PERFORM pg_advisory_lock(receiver_id);
+
+	-- Deduct the amount from the sender's account
+	UPDATE accounts
+	SET balance = balance - transfer_amount
+	WHERE account_id = sender_id;
+
+	-- Add the amount to the receiver's account
+	UPDATE accounts
+	SET balance = balance + transfer_amount
+	WHERE account_id = receiver_id;
+
+	-- Release the advisory locks
+	PERFORM pg_advisory_unlock(sender_id);
+	PERFORM pg_advisory_unlock(receiver_id);
+
+END;
+$$;
+
+parsed json -
+	{
+	  "PLpgSQL_function": {
+	    "datums": [
+	      ...
+	    ],
+	    "action": {
+	      "PLpgSQL_stmt_block": {
+	        "lineno": 2,
+	        "body": [
+	          {
+	            "PLpgSQL_stmt_perform": {
+	              "lineno": 4,
+	              "expr": {
+	                "PLpgSQL_expr": {
+	                  "query": "SELECT pg_advisory_lock(sender_id)",
+	                  "parseMode": 0
+	                }
+	              }
+	            }
+	          },
+	          {
+				... similar to above
+	          },
+	          {
+	            "PLpgSQL_stmt_execsql": {
+	              "lineno": 8,
+	              "sqlstmt": {
+	                "PLpgSQL_expr": {
+	                  "query": "UPDATE accounts \n    SET balance = balance - transfer_amount \n    WHERE account_id = sender_id",
+	                  "parseMode": 0
+	                }
+	              }
+	            }
+	          },
+	          {
+				.... similar to above
+	          },
+	          {
+	            ... similar to below
+	          },
+	          {
+	            "PLpgSQL_stmt_perform": {
+	              "lineno": 19,
+	              "expr": {
+	                "PLpgSQL_expr": {
+	                  "query": "SELECT pg_advisory_unlock(receiver_id)",
+	                  "parseMode": 0
+	                }
+	              }
+	            }
+	          },
+	          {
+	            "PLpgSQL_stmt_return": {}
+	          }
+	        ]
+	      }
+	    }
+	  }
+	}
+*/
+func TraversePlPgSQLActions(action interface{}, plPgSqlStatements *[]string) {
```

Comment:
how about calling the argument `fieldValue` instead of actions, because it's only actions for the first time, after that it receursively traverses through all the fields, correct?

Reply:
yes